proxy/apifilterproxy/filters/auth/rancher_token_validate_filter.go

```diff
@@ -0,0 +1,289 @@
+package auth
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/rancher/websocket-proxy/proxy/apifilterproxy/filters"
+	"github.com/rancher/websocket-proxy/proxy/apifilterproxy/model"
+)
+
+const (
+	name = "authTokenValidator"
+)
+
+//AuthorizeData is for the JSON output
+type AuthorizeData struct {
+	Message string `json:"message,omitempty"`
+}
+
+//MessageData is for the JSON output
+type MessageData struct {
+	Data []interface{} `json:"data,omitempty"`
+}
+
+func init() {
```

Comment:
Quick pseudo code, your mileage may very:
```
func InitHandler(filterConfigFile string, cattleAddr string) (*APIFiltersHandler, bool) {
    filters := getFilters() 
    manager.SetConfig(filterConfigFile, cattleAddr)
    router := newFilterRouter(manager.ConfigFields, filters)
    
    filterHandler = &APIFiltersHandler{filterRouter: router}
    return filterHandler, manager.APIFilterProxyReady
}

func getFilters() map[string]handler {
   m // handler map
  authHandler := auth.NewHandler()
  m[authHandler.getName()] = authhandler
  
  // repeat for other handlers.
}
```
I realize this seems like you are creating a dependency between this package and each of the filters, but you already had the dependency, you were just obfuscating it through the anonymous import